grid/crm/contacts/maps/pipedrive.suma

```diff
@@ -0,0 +1,197 @@
+profile = "crm/contacts@1.0"
+provider = "pipedrive"
+
+map Create {
+
+    set {
+        body = {
+            name: [input.firstName, input.lastName].join(" "),
+            email: [{
+                label: "",
+                value: input.email,
+                primary: true,
+
+            }],
+            phone: [{
+                label: "",
+                value: input.phone,
+                primary: true,
+
+            }]
+        }
+    }
+
+    organization = call GetOrCreateOrganizatio(input = input)
+    set if(organization){
+        body.org_id = organization.id
+    }
+
+    http POST "/persons" {
+        security "api-key"
+
+        request "application/json" {
+            body = body
+        }
+
+        response 201 "application/json" {
+            map result {
+                id = body.data.id
+            }
+        }
+
+        response {
+            error = call MapError(statusCode = statusCode, body = body)
+            map error error
+        }
+    }
+}
+
+map Update {
+    // todo
+}
+map Search {
+
+    set {
+        FIELD_MAP = {
+            firstName: 'name',
+            lastName: 'name'
+        }
+    }
+
+    // https://developers.pipedrive.com/docs/api/v1/Persons#searchPersons
+    http GET "/persons/search" {
+        security "api-key"
+
+        request "application/json" {
+            query {
+                // operator does not exist
+                term = input.value,
+                fields = [FIELD_MAP[input.property] || input.property]
+            }
+        }
+
+
+        response 200 "application/json" {
+            map result body.data.items.map(({item}) =>
+                ({
+                    id: item.id,
+                    email: item.primary_email,
+                    phone: item.phones[0],
+                    // todo replace first name and lastname with name
+                    company: item.organization && item.organization.name,
+                    country: null,
+                    customProperties: item.custom_fields,
+                })
+            )
+        }
+
+        response {
+            error = call MapError(statusCode = statusCode, body = body)
+            map error error
+        }
+    }
+}
+
+operation GetOrCreateOrganizatio {
+
+    data = call GetOrganizationByName(input = args.input)
```

Comment:
This way of calling ends up silently throwing away the error if the operation call fails, see:

https://superface.ai/docs/comlink/reference/map#calling-operation

This may be useful in some cases - but here it may result in e.g. creating duplicate organizations, in case the operation falls intermittently, or due to some other reason.

It'd be best to use the "Calling with a handler" syntax (documented in the link above)